# Intro to Pandas

# Contents
1. Import libraries
2. Import datasets
3. Check imported data 

# 1. Import libraries

In [5]:
#Import libraries
import pandas as pd
import numpy as np
import os

# 2. Import datasets

In [7]:
#Define the project path
path = r'/Users/davidgriesel/Documents/0 - CareerFoundry/02 - Data Analytics Immersion/04 - Python Fundamentals for Data Analysts/Instacart Basket Analysis - IC 202409'

### 2.1. Orders

In [9]:
#Define a list with column names, excluding the “eval_set” column
vars_list = ['order_id', 'user_id', 'order_number', 'order_dow', 'order_hour_of_day', 'days_since_prior_order']

In [10]:
#Display the list
vars_list

['order_id',
 'user_id',
 'order_number',
 'order_dow',
 'order_hour_of_day',
 'days_since_prior_order']

In [11]:
#Import dataset with columns limited to those on the list
df_orders = pd.read_csv(os.path.join(path, '02 - Data', 'Original Data', 'orders.csv'), usecols = vars_list)

### 2.2. Products

In [13]:
#Import the dataset
df_products = pd.read_csv(os.path.join(path, '02 - Data', 'Original Data', 'products.csv'))

# 3. Check imported data

## 3.1. Orders

### 3.1.1. head()

In [17]:
#Display the first 5 rows of the dataframe
df_orders.head()

order_id  user_id  order_number  order_dow  order_hour_of_day  \
0   2539329        1             1          2                  8   
1   2398795        1             2          3                  7   
2    473747        1             3          3                 12   
3   2254736        1             4          4                  7   
4    431534        1             5          4                 15   

   days_since_prior_order  
0                     NaN  
1                    15.0  
2                    21.0  
3                    29.0  
4                    28.0

## 3.2. Products

### 3.2.1. head (specified)

In [20]:
#Display the first 20 rows of the dataframe
df_products.head(20)

product_id                                       product_name  aisle_id  \
0            1                         Chocolate Sandwich Cookies        61   
1            2                                   All-Seasons Salt       104   
2            3               Robust Golden Unsweetened Oolong Tea        94   
3            4  Smart Ones Classic Favorites Mini Rigatoni Wit...        38   
4            5                          Green Chile Anytime Sauce         5   
5            6                                       Dry Nose Oil        11   
6            7                     Pure Coconut Water With Orange        98   
7            8                  Cut Russet Potatoes Steam N' Mash       116   
8            9                  Light Strawberry Blueberry Yogurt       120   
9           10     Sparkling Orange Juice & Prickly Pear Beverage       115   
10          11                                  Peach Mango Juice        31   
11          12                         Chocolate Fudge Layer Cake       119   
12          13                                  Saline Nasal Mist        11   
13          14                     Fresh Scent Dishwasher Cleaner        74   
14          15                           Overnight Diapers Size 6        56   
15          16                      Mint Chocolate Flavored Syrup       103   
16          17                                  Rendered Duck Fat        35   
17          18                Pizza for One Suprema  Frozen Pizza        79   
18          19   Gluten Free Quinoa Three Cheese & Mushroom Blend        63   
19          20     Pomegranate Cranberry & Aloe Vera Enrich Drink        98   

    department_id  prices  
0              19     5.8  
1              13     9.3  
2               7     4.5  
3               1    10.5  
4              13     4.3  
5              11     2.6  
6               7     4.4  
7               1     1.1  
8              16     7.0  
9               7     8.4  
10              7     2.8  
11              1     9.4  
12             11     1.1  
13             17     6.5  
14             18    11.2  
15             19     5.2  
16             12    17.1  
17              1    10.6  
18              9    12.0  
19              7     6.0

### 3.2.2. tail (specified)

In [22]:
#Display the last 35 rows of the dataframe
df_products.tail(35)

product_id                                       product_name  \
49658       49654                   Teriyaki Sauce, Sesame, Original   
49659       49655                                        Apple Cider   
49660       49656                         Masada Kosher Pocket Bread   
49661       49657                                  Cabernet Tomatoes   
49662       49658                         Brie with Herbs Foil Wedge   
49663       49659                            Organic Creamed Coconut   
49664       49660                        Professionals Sleek Shampoo   
49665       49661                                              Porto   
49666       49662                       Bacon Cheddar Pretzel Pieces   
49667       49663  Ultra Protein Power Crunch Peanut Butter N' Ho...   
49668       49664                     Lemon Cayenne Drinking Vinegar   
49669       49665      Super Dark Coconut Ash & Banana Chocolate Bar   
49670       49666                      Ginger Snaps Snacking Cookies   
49671       49667           Enchilada with Spanish Rice & Beans Meal   
49672       49668                     Apple Cinnamon Scented Candles   
49673       49669                                   K Cup Dark Blend   
49674       49670                                Beef Summer Sausage   
49675       49671                               Milk Chocolate Drops   
49676       49672                             Cafe Mocha K-Cup Packs   
49677       49673              Stone Baked Multi Grain Artisan Rolls   
49678       49674            Frozen Greek Yogurt Bars Chocolate Chip   
49679       49675                Cinnamon Dolce Keurig Brewed K Cups   
49680       49676                             Ultra Red Energy Drink   
49681       49677                    Thick & Chunky Sloppy Joe Sauce   
49682       49678                    Large Chicken & Cheese Taquitos   
49683       49679                            Famous Chocolate Wafers   
49684       49680                 All Natural Creamy Caesar Dressing   
49685       49681            Spaghetti with Meatballs and Sauce Meal   
49686       49682                                 California Limeade   
49687       49683                                     Cucumber Kirby   
49688       49684          Vodka, Triple Distilled, Twist of Vanilla   
49689       49685                 En Croute Roast Hazelnut Cranberry   
49690       49686                                   Artisan Baguette   
49691       49687         Smartblend Healthy Metabolism Dry Cat Food   
49692       49688                             Fresh Foaming Cleanser   

       aisle_id  department_id  prices  
49658         5             13     4.0  
49659        98              7    10.7  
49660       128              3     7.1  
49661        83              4     8.3  
49662         2             16     3.9  
49663        17             13     3.1  
49664        22             11     6.7  
49665       134              5     8.2  
49666       107             19     3.6  
49667        57             14    10.2  
49668       100             21    13.7  
49669        45             19     6.9  
49670        61             19     5.2  
49671        38              1     6.6  
49672       101             17     5.6  
49673       100             21     4.7  
49674       106             12    19.2  
49675        45             19     3.0  
49676        26              7     6.5  
49677       129              1     5.6  
49678        37              1    11.1  
49679        26              7    14.0  
49680        64              7    14.5  
49681        59             15     8.9  
49682       129              1     3.4  
49683        61             19     6.0  
49684        89             13     4.9  
49685        38              1     6.9  
49686        98              7     4.3  
49687        83              4    13.2  
49688       124              5     5.3  
49689        42              1     3.1  
49690       112              3     7.8  
49691        41     

### 3.2.3. columns

In [24]:
#Return the column names of the dataframe
df_products.columns

Index(['product_id', 'product_name', 'aisle_id', 'department_id', 'prices'], dtype='object')

### 3.2.4. shape

In [26]:
#Get the dimensions of the dataframe
df_products.shape

(49693, 5)

### 3.2.5. describe()

In [28]:
#Generate descriptive statistics for numerical columns in the dataframe
df_products.describe()

product_id      aisle_id  department_id        prices
count  49693.000000  49693.000000   49693.000000  49693.000000
mean   24844.345139     67.770249      11.728433      9.994136
std    14343.717401     38.316774       5.850282    453.519686
min        1.000000      1.000000       1.000000      1.000000
25%    12423.000000     35.000000       7.000000      4.100000
50%    24845.000000     69.000000      13.000000      7.100000
75%    37265.000000    100.000000      17.000000     11.200000
max    49688.000000    134.000000      21.000000  99999.000000

##### The maximum value of the aisle_id column is 134

### 3.2.6. dtypes

In [31]:
#Return the data types of each column in the dataframe
df_products.dtypes

product_id         int64
product_name      object
aisle_id           int64
department_id      int64
prices           float64
dtype: object

##### The data type of the department_id column is int64

# Task

## 1. Download data
Download the data file containing your project data sets for this Exercise, extract it, and move the data files into your “Original Data” folder.

##### Data downloaded

## 2. Create a notebook
Create a notebook in Jupyter for this Exercise in your “Scripts” folder and give it a descriptive title. Be sure to set up section headings and use comments in your notebook as you proceed with the steps in the rest of the task.

##### Notebook created

## 3. Import libraries
Import the pandas, NumPy, and os libraries.

##### Refer: 1. Import libraries

## 4. Create project path
Turn your project folder path into a string as outlined in the Exercise text.

##### Refer: 2. Import datasets

## 5. Import 'orders' dataset
Import the “orders.csv” data set into your notebook using the os library and the vars_list shortcut, omitting the “eval_set” column in your final dataframe.

##### Refer: 2.1 Orders

## 6. Print head
Print the first 5 rows of your dataframe.

##### Refer: 3.1.1 head()

## 7. Import 'products' dataset
Import the “products.csv” file into Jupyter as df_prods (as opposed to df, which is what your “orders.csv” file is named in Jupyter). You can use the path variable you created in step 4 to import “products.csv” with the os library.

##### Refer: 2.2 Products

## 8. Print head and tail
Print the first 20 rows and the last 35 rows of the df_prods dataframe.

##### Refer: 3.2.1 head (specified)
##### Refer: 3.2.2 tail (specified)

## 9. Print column names
Print the names of the columns in the df_prods dataframe.

##### Refer: 3.2.3 columns

## 10. Print dimensions
Print the number of rows and columns in the df_prods dataframe.

##### Refer: 3.2.4. shape

## 11. Print descriptive statistics and data types
Answer the following questions using the relevant commands. Use markdown mode for your Jupyter cells to provide written answers within your Jupyter notebook after executing the commands:
- What’s the maximum value of the “aisle_id” column?
- What’s the data type of the “department_id” column?

##### Refer: 3.2.5 describe() 
##### Refer: 3.2.6 dtypes

## 12. Save and submit notebook
Save your notebook, then locate it in your file explorer, compress it, and upload the compressed ZIP file here for your tutor to review.

##### Notebook saved and submitted